torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2393,11 +2390,11 @@ Tensor svd_backward(const std::vector<torch::autograd::Variable> &grads, const T
 
   Tensor sigma_term;
   if (gsigma.defined()) {
-    gsigma = gsigma.to(self.dtype());
+    gsigma = gsigma.to(U.dtype());
     // computes u @ diag(gsigma) @ vh
     sigma_term = at::matmul(u * gsigma.unsqueeze(-2), vh);
   } else {
-    sigma_term = at::zeros_like(self, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+    sigma_term = at::zeros_like(m >= n ? u : vh, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
```

Comment:
sigma is going to be a full-sized zero matrix then. Maybe we can just init it with a zero scalar and rely on broadcasting?

Reply:
This line is gone in the next PR in the stack in favour of a better rewrite of the funciton. Here I just changed it to avoid having this function depending on `self`, as it was not necessary.